code401challenges/src/main/java/code401challenges/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,75 @@
+package code401challenges.stacksandqueues;
+
+import java.util.NoSuchElementException;
+
+public class Queue {
+    // This object should be aware of a default empty value assigned to front when the queue is created.
+
+    public Node front;
+    public Node back;
+
+    // this is a constructor
+    public Queue(){
+        front = null;
+    }
+
+    //Define a method called isEmpty that does not take an
+    // argument, and returns a boolean indicating whether
+    // or not the queue is empty.
+
+    public boolean isEmpty(){
+        return front == null;
+    }
+
+    //Define a method called enqueue which takes any value as
+    // an argument and adds a new node with that value to the
+    // back of the queue with an O(1) Time performance.
+
+    public void enqueue(int value){
+        Node runner =  new Node(value);
+        if(isEmpty()){
+            front = back = runner;
+        }
+        else if (front == back){
+            front.next = runner;
+            back = runner;
+        }
+        else{
+            back.next = runner;
+            back = runner;
+        }
+    }
+
+    //Define a method called dequeue that does not take any
+    //argument, removes the node from the front of the
+    // queue, and returns the node’s value.
+    public int dequeue(){
+        if(isEmpty()){
+            throw new NoSuchElementException("sorry This queue is empty");
+        }
+        else if(front == back){
+            Node runner = front;
```

Comment:
Same here, `runner` implies it is being used to iterate through the Queue, but it's just a reference to the node we are taking out of the queue